aten/src/ATen/native/cuda/DilatedMaxPool2d.cu

```diff
@@ -355,27 +357,44 @@ void max_pool2d_with_indices_out_cuda_template(
           const int max_threads = std::min<int>(
               at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock, CUDA_MAX_THREADS);
           int* maxThreadsDim = at::cuda::getCurrentDeviceProperties()->maxThreadsDim;
-          int block_x = std::min<int>(
-              maxThreadsDim[0], std::min<int>(lastPow2(nInputPlane), at::cuda::warp_size()));
-          int block_y = std::min<int>(
-              maxThreadsDim[1], std::min<int>(lastPow2(outputWidth), max_threads / block_x));
-          int block_z = std::min<int>(
-              maxThreadsDim[2], std::min<int>(lastPow2(outputHeight), max_threads / block_x / block_y));
-          block_x = std::min<int>(
-              maxThreadsDim[0], std::min<int>(lastPow2(nInputPlane), max_threads / block_y / block_z));
-          const dim3 block(block_x, block_y, block_z);
-          int grid_x = nbatch;
-          int grid_y = cuda::ATenCeilDiv(safe_downcast<int, int64_t>(outputWidth), block_y*BLOCK_STRIDE);
```

Comment:
We shouldn't need multiple kernel launching, as we are striding along width/height in the kernel already.

I would discard all changes and simply add change this line (and grid_z as well)
```
int grid_y = std::min<int>(
at::cuda::getCurrentDeviceProperties()->maxGridSize[1],
cuda::ATenCeilDiv(safe_downcast<int, int64_t>(outputWidth), block_y*BLOCK_STRIDE));
``` 

Please do *NOT* do the same thing for grid_x, as the kernel does not loop over batch dimension. The fix you have here could be used for that.

Reply:
I updated grid, and removed slicing, but I still see the performance degradation for backward function.

Reply:
hmmm, perf regression doesn't really make any sense here. It's basically the same kernel.
Not sure if it's related to that typo I pointed out in the other comment.

Changing the grid size would affect cache hit? I need to think about it further before I can make any decisive call. But let's fix the typo and hope the problem goes away